compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -525,6 +541,54 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 },
             );
         }
+
+        if !need_traits_migrations.is_empty() {
+            let (migration_string, migrated_variables_concat) =
```

Comment:
I'd create a function or clousre that does most of this, the two codeblocks are basically the same with the paramters being the lint and the associated messages